community/gardening/dev-environment.md

```diff
@@ -0,0 +1,223 @@
+
+
+
+## Kubernetes and Docker method:
+
+### Kubernetes installation methods
+_The instructions for this method are in beta. Pull requests welcome!_
+
+1. Install Spinnaker to a Kubernetes cluster. There are several ways to do this:
+   - [Install Halyard in Docker](/setup/install/halyard/#install-halyard-on-docker)
+     - In your `docker run` command, mount the `.kube` directory to the container to allow you to modify `.kube` config files on your local machine and persist the changes inside the container:
+       ```
+       docker run -p 8084:8084 -p 9000:9000 \
+         --name halyard --rm \
+         -v ~/.hal:/home/spinnaker/.hal \
+         -v ~/.kube:/home/spinnaker/.kube \
+         -it \
+         gcr.io/spinnaker-marketplace/halyard:stable
+       ```
+   - Use the [Spinnaker for Google Cloud Engine](https://cloud.google.com/docs/ci-cd/spinnaker/spinnaker-for-gcp) solution, which installs Spinnaker to Google Kubernetes Engine.
+   - Consult the installation guide for [Amazon Kubernetes Service](https://aws.amazon.com/blogs/opensource/continuous-delivery-spinnaker-amazon-eks/)
+   - Use the new [Spinnaker Operator](https://docs.armory.io/spinnaker/operator/#install-operator) to quickly install with `kubectl` commands.
+     - Install the Operator in [cluster mode](https://docs.armory.io/spinnaker/operator/#installing-operator-in-cluster-mode)
+     - Front50 won't start up successfully until you point Spinnaker to persistent storage, such as an S3 bucket. Update `deploy/spinnaker/basic/SpinnakerService.yml` as in this snippet:
+    ```
+    ...
+    spinnakerConfig:
+      config:
+        persistentStorage:
+          persistentStoreType: s3
+          s3:
+            bucket: mybucket
+            rootFolder: front50
+        version: 2.18.0
+    ...
+    ```
+
+2. Open a bash shell in the location where Halyard is installed.
+  - If Halyard is running in Docker, run `docker exec -it halyard bash` to enter a shell.   
+3. Edit the Kubernetes block of your .hal/config with your namespace and kubeconfig file location to enable a Kubernetes install, as in this snippet:
+  ```
+  ...
+  kubernetes:
+    enabled: true
+    accounts:
+      name: kubernetes
+      requiredMembership: []
+      providerVersion: V2
+      permissions: []
+      dockerRegistries: []
+      configureImagePullSecrets: true
+      cacheThreads: 1
+      namespaces:
+        - <namespace>
+      omitNamespaces: []
+      kinds: []
+      omitKinds: []
+      customResources: []
+      cachingPolicies: []
+      kubeconfigFile: </path/to/kubeconfig/>
+      oAuthScopes: []
+      onlySpinnakerManaged: false
+    primaryAccount: kubernetes
+      ...
+  ```
+4. Port-forward the externally-hosted Spinnaker services to your local machine
+  - You may use [NGROK](https://ngrok.com/download)
+  - Or, try this Fish function: `pf-spinnaker` loops through all of the Spinnaker services in your Kubernetes namespaces and forwards their ports to your local machine. Try it:
```

Comment:
Where did this come from? I'm a happy fish user, but it's a pretty niche shell... I wouldn't think most people would have even heard of it, let alone know how to use it? I suppose it's fine for now, but we should probably translate this to bash at some point...

Reply:
@dnilasor can take a pass at this as she iterates.